packages/react-form/src/hooks/list/hooks/reducer.ts

```diff
@@ -211,6 +233,25 @@ function reduceList<Item extends object>(
         list: [...state.list, ...action.payload.list.map(initialListItemState)],
       };
     }
+    case 'editFieldItem': {
+      const {editedItem, index} = action.payload;
+
+      const newList = [...state.list];
+      const [item] = newList.splice(index, 1);
+
+      // Don't do a full rewrite of the object, only edit data which is supplied.
+      const newItem = {
+        ...item,
+        ...editedItem,
+      };
+
+      newList[index] = newItem;
+
+      return {
+        ...state,
+        list: newList,
+      };
+    }
```

Comment:
`state.list` will be mutated regardless here per JavaScripts (weird) handling of objects.  [Example here](https://screenshot.click/02-04-1f214-j5tzh.png) and [explanation here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice).

It's unlikely this will cause an issue in its implementation but could cause side effects if someone may need to use this in an unorthodox way (for example, in a loop or asynchronous action).

A safer approach might be

```tsx
const {editedItem, index} = action.payload;
const list = state?.list?.map?.((item, i) => i === index ? { ...item, ...editedItem } : item);
return {
  ...state,
 list
}
```